iceoryx_hoofs/include/iceoryx_hoofs/log/logstream.hpp

```diff
@@ -14,158 +14,115 @@
 
-// helper struct for SFINAE of LogStream& operator<<
-struct LogBin
-{
+  private:
+    T m_value;
 };
 
-struct LogBin8 : private LogBin
-{
-    uint8_t value;
-    constexpr LogBin8(uint8_t value) noexcept
-        : value(value)
-    {
-    }
-};
-struct LogBin16 : private LogBin
-{
-    uint16_t value;
-    constexpr LogBin16(uint16_t value) noexcept
-        : value(value)
-    {
-    }
-};
-struct LogBin32 : private LogBin
-{
-    uint32_t value;
-    constexpr LogBin32(uint32_t value) noexcept
-        : value(value)
-    {
-    }
-};
-struct LogBin64 : private LogBin
-{
-    uint64_t value;
-    constexpr LogBin64(uint64_t value) noexcept
-        : value(value)
-    {
-    }
-};
-struct LogRawBuffer
+template <typename T, typename = std::enable_if_t<std::is_arithmetic<T>::value>>
+inline constexpr LogHex<T> hex(const T value) noexcept;
+
+inline LogHex<uint64_t> hex(const void* const ptr) noexcept;
+
+template <typename T>
+class LogOct
 {
-    const uint8_t* data;
-    uint8_t size;
+  public:
+    friend class LogStream;
+
+    template <typename = std::enable_if_t<std::is_arithmetic<T>::value>>
+    inline explicit constexpr LogOct(const T value) noexcept;
+
+  private:
+    T m_value;
 };
 
-class Logger;
+template <typename T, typename = std::enable_if_t<std::is_arithmetic<T>::value>>
+inline constexpr LogOct<T> oct(const T value) noexcept;
 
 class LogStream
 {
   public:
-    LogStream(Logger& logger, LogLevel logLevel = LogLevel::kWarn) noexcept;
-
-    virtual ~LogStream() noexcept;
-
-    void Flush() noexcept;
-
-    LogStream& operator<<(const char* cstr) noexcept;
-
-    LogStream& operator<<(const std::string& str) noexcept;
-
-    template <typename T, typename std::enable_if<std::is_arithmetic<T>::value, int>::type = 0>
-    LogStream& operator<<(const T val) noexcept
-    {
-        m_logEntry.message.append(cxx::convert::toString(val));
-        m_flushed = false;
-        return *this;
-    }
-
-    template <typename T, typename std::enable_if<std::is_base_of<LogHex, T>::value, int>::type = 0>
-    LogStream& operator<<(const T val) noexcept
-    {
-        std::stringstream ss;
-        // the '+val' is there to not interpret the uint8_t as char and print the character instead of the hex value
-        ss << "0x" << std::hex << +val.value;
-        m_logEntry.message.append(ss.str());
-        m_flushed = false;
-        return *this;
-    }
-
-    template <typename T, typename std::enable_if<std::is_base_of<LogBin, T>::value, int>::type = 0>
-    LogStream& operator<<(const T val) noexcept
-    {
-        m_logEntry.message.append("0b");
-        m_logEntry.message.append(std::bitset<std::numeric_limits<decltype(val.value)>::digits>(val.value).to_string());
-        m_flushed = false;
-        return *this;
-    }
-
-    LogStream& operator<<(const LogRawBuffer& value) noexcept;
+    inline LogStream(const char* file, const int line, const char* function, LogLevel logLevel) noexcept;
```

Comment:
The doxygen documentation will be completed with a follow up PR which will be available before this PR will be merged